internal/provider/provider.go

```diff
@@ -164,35 +164,48 @@ func keyFromData(rootData map[string]interface{}, secretKey string) (string, err
 		data = rootData
 	}
 
-	content, ok := data[secretKey].(string)
-	if !ok {
-		return "", fmt.Errorf("failed to get secret content %q as string", secretKey)
+	// Special-case the most common format of strings so the contents are
+	// returned plainly without quotes that json.Marshal would add.
+	if content, ok := data[secretKey].(string); ok {
+		return []byte(content), nil
+	}
+
+	// Arbitrary data can be returned in the data field of an API secret struct.
+	// It's already been Unmarshalled from the response, so in theory,
+	// marshalling should never realistically fail, but don't log the error just
+	// in case, as it could contain secret contents if it does somehow fail.
+	if content, err := json.Marshal(data[secretKey]); err == nil {
+		return content, nil
 	}
```

Comment:
Interesting, I naively thought these would be flipped (checking for json and then string), but of course data is already unmarshalled here.

Reply:
Yeah it took me a surprising amount of iteration to arrive at this. I wanted to just have JSON, but then the extra quotes ruin the most common usage.